open_discussions_api/channels/client_test.py

```diff
@@ -0,0 +1,76 @@
+"""Tests for user client API"""
+import json
+
+import pytest
+
+
+def test_create_channel(api_client):
+    """test a happpy path"""
+    resp = api_client.channels.create(
+        title="new test channel",
+        name="examplechannel",
+        public_description="a good channel for all things",
+        channel_type="public"
+    )
+    assert resp.status_code == 201
+    assert json.loads(resp.request.body) == {
+        "title": "new test channel",
+        "name": "examplechannel",
+        "public_description": "a good channel for all things",
+        "channel_type": "public"
+    }
+    assert resp.json() == {
+        "title": "new test channel",
+        "name": "examplechannel",
+        "public_description": "a good channel for all things",
+        "channel_type": "public"
+    }
+
+
+def test_create_private_channel(api_client):
+    """test a different happy path"""
+    resp = api_client.channels.create(
+        title="new private channel",
+        name="privatechannel",
+        public_description="a good channel for all things",
+        channel_type="private"
+    )
+    assert resp.status_code == 201
+    assert json.loads(resp.request.body) == {
+        "title": "new private channel",
+        "name": "privatechannel",
+        "public_description": "a good channel for all things",
+        "channel_type": "private"
+    }
+    assert resp.json() == {
+        "title": "new private channel",
+        "name": "privatechannel",
+        "public_description": "a good channel for all things",
+        "channel_type": "private"
+    }
+
+
+def test_create_channel_with_no_arguments(api_client):
+    """should raise when provided nothing"""
+    with pytest.raises(AttributeError) as err:
+        api_client.channels.create()
+    assert str(err.value) == "No fields provided to create the channel"
```

Comment:
Note: I personally believe that it is a bad idea to assert on the exception messages, but feel free to leave as is if you think it is needed.

Reply:
Why do you think it's a bad idea? I was just trying to make sure that the correct error was raised here, if there's a better way to do so I can definitely change it.

Reply:
I think it is a bad idea because the messages are not a clear indication of the semantic of an error.
You are already asserting that the error is correct when you check that the raised exception is `AttributeError`. If you need to be more specific about the exception, it is probably a better idea to create exceptions that subclass `AttributeError` and that are more specific about what went wrong.

Reply:
ok, I'll make some error subclasses then

Reply:
great

Reply:
I added some new error classes subclassing `AttributeError`, and I removed one of the `err.value` assertions. I kept the ones where I'm passing a formatted error message, just to test that the formatting is what I expect.